qiskit_ibm_provider/job/ibm_circuit_job.py

```diff
@@ -625,7 +637,8 @@ def refresh(self) -> None:
         self._client_version = self._extract_client_version(
             api_response.pop("client_info", None)
         )
-        self._set_result(api_response.pop("result", None))
+        api_result = self._runtime_client.job_results(self.job_id())
```

Comment:
What happens if the job is not done? Does this throw an error?

Reply:
It *should* set the results to `None` and move on, but I don't see it happening in the code. I'll test and fix this.

Reply:
In `qiskit-ibm-runtime` getting results is only called once the job is complete (otherwise the server throws an error). So I did the same in `refresh` (the only place where it might have been called before the job is done)